tools/generate_json_map_class.js

```diff
@@ -1005,27 +1420,110 @@ const generateStructHeader = (dirname, filename, json_list) => {
 };
 
 
+// ----------------------------------------------------------------------------
+// generate typescript data file function
+// ----------------------------------------------------------------------------
+/**
+ * generateTsData
+ * @param dirname {string} - directory path.
+ * @param filename {string} - output file name.
+ * @param jsonClassMap {Object} - class object map.
+ * @param jsonTypeList {string[]} - sorted class map keys.
+ * @param functionList {string[]} - function names.
+ */
+const generateTsData = (dirname, filename, jsonClassMap, jsonTypeList, functionList) => {
+  const result = [];
+  const processedStructTypes = [];
+
+  let path = `${dirname}/${filename}`;
+  if (path.startsWith(__dirname)) {
+    path = path.substr(__dirname.length);
+  }
+  while (path.indexOf('/', 0) == 0) {
+    path = path.substr(1);
+  }
+  while (path.indexOf('//') >= 0) {
+    path = path.replace('//', '/');
+  }
+
+  // initialize
+  const project = new Project({
+      tsConfigFilePath: `${__dirname}/../tsconfig.json`,
+      addFilesFromTsConfig: false,
+  });
+
+  // add source files
+  // const namespaceFile = project.createSourceFile(path, 'export namespace Cfd {}\n');
```

Comment:
There are also quite some commented out code. If there is a reason to keep it, nice to put a comment about why, otherwise better to delete.

Reply:
There is no reason to keep anything but the code for debugging, I was delete it. Thank you very match!